homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -94,39 +97,41 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
                 hass.data[D_EGARDIASRV] = egardiaserver.EgardiaServer('',
                                                                       rs_port)
                 bound = hass.data[D_EGARDIASRV].bind()
+
                 if not bound:
                     raise IOError("Binding error occurred while " +
                                   "starting EgardiaServer")
+                hass.data[D_EGARDIASRV].register_callback(
+                    eg_dev.handle_status_event)
+                hass.data[D_EGARDIASRV].start()
         except IOError:
             return False
 
-    add_devices([EgardiaAlarm(
-        name, egardiasystem, hass, rs_codes)], True)
+    def handle_stop_event(event):
+        hass.data[D_EGARDIASRV].stop()
+
+    # listen to home assistant stop event
+    hass.bus.listen('homeassistant_stop', handle_stop_event)
```

Comment:
Use `hass.bus.listen_once` and import the home assistant stop event constant from const.py and use that here.